context/PokemonState.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useReducer } from 'react';
+import PokemonContext from './pokemonContext';
+import PokemonReducer from './pokemonReducer';
+import { Pokemon } from '../api/pokemon';
+import { SAVE_FAV_POKEMON } from './types';
+
+export interface State {
+  favPokemon: Pokemon | null;
+}
+
+export type Action = { type: 'SAVE_FAV_POKEMON'; payload: any };
```

Comment:
you should use `SAVE_FAV_POKEMON` constant here as well, but generally I'd make it more Pokemon-related:
```ts
export interface PokemonState {
...
}

type PokemonActionType = SAVE_FAV_POKEMON; // you can easily extend it by adding | ANOTHER | ONE_MORE;

export type PokemonAction = { type: PokemonActionType; payload: any };
```
Technically you could also type `payload`, but it's more complex and easier to do correctly when you have more than one action (as you can see "the whole picture" then), so let's keep it as `any` for now